Bar j

bar-jakubovitz opened the pull request:
Add implementation for Nurego::Offering.as_cf_catalog

anatb commented on lib/nurego/offering.rb:
```diff
@@ -10,8 +11,63 @@ def self.current(params = {}, api_key = nil)
       Util.convert_to_nurego_object(response, api_key)
     end
 
+    def self.as_cf_catalog(url, api_key )
```
space before )

anatb replied:
is your request different than retrieve? because if it isn't then it's more oo if you load an offering and then have the parsing on the offering object instance, not the offering class.
also - this is a public library - meaning the code is visible to the world which doesn't know what cf is - please use full name in the API.

anatb commented on lib/nurego/offering.rb:
```diff
@@ -10,8 +11,63 @@ def self.current(params = {}, api_key = nil)
+          offer_id: nurego_offering['id'],
+          offer_name: nurego_offering['name'],
+          offer_description: nurego_offering['description'],
+          :services => []
```
consistency within the same hash?

anatb commented:
in theory - need to add a test when you add a code. in reality - specs are not excellent here. I would however add one - so when we fix them we don't forget to test this use case. also - if the distribution channel is the problem - override it for the test - and check the parser. (or just create unit tests for the parser, but preferably both).

bar-jakubovitz commented:
@anatb please recheck, i updated the remarks and added the test suit you asked. (there are 2 integration tests failing: passwordreset and update organization subscription)
and the new test should fail until we merge my changes to BSS (exposes services id name a description for offering)

anatb commented on lib/nurego/offering.rb:
```diff
@@ -10,8 +11,63 @@ def self.current(params = {}, api_key = nil)
 
+    def to_cloud_foundry_catalog
+      # require BSS to expose service_id, service_name, service_description for Offering
+      Offering.offering_to_broker_catalog(self).to_json
```
not what I expected. was expecting self.to_broker_catalog. you're in an offering class. all you have to do is make offering_to_broker_catalog instance and not class method (and rename it).

anatb commented:
other than my last comment this is ok. please don't merge to master anything you're not ready to send to prod next week.

anatb commented:
@ravityunian - check this out

bar-jakubovitz commented:
@anatb Added the service endpoint to nurego-ruby (for catalog of broker)
also fixed and tests added
removed the CF integration from offering (now is in service)
added registration find by external id and added tests

anatb commented on lib/nurego/service.rb:
```diff
@@ -0,0 +1,56 @@
+    private
+
+    def service_to_cloud_foundry_catalog
+      cf_catalog = {
```
what happens if there's no offering?

bar-jakubovitz replied:
if there is no service you cant get a Nurego::Service class
is there a way to have a service with no offering ?
i suspect a no method exception will happen if you try  parse a service with no offer. (untested)
what do we want to happen in such a case ? return an empty catalog ? raise error ?

anatb commented on spec/integration/registrations_spec.rb:
```diff
@@ -2,25 +2,63 @@
+        Nurego::Registration.find_by_external_id('duplicated_external_instance_id_test')
+      rescue Nurego::InvalidRequestError => e
+        e.http_status.should eq(400)
+        !!e.http_body.should(be_true)
```
what's !! ? new to me.

anatb commented on spec/integration/services_spec.rb:
```diff
@@ -0,0 +1,68 @@
+  end
+
+  it "can fetch a service " do
+    service = Nurego::Service.retrieve("197f9207-7325-4d75-983a-116b43fb9cc8")
```
this will only run on your local system, right?

bar-jakubovitz replied:
this is the nurego service GUID from the nurego init so i think its supposed to run anywhere.

anatb commented on spec/integration/services_spec.rb:
```diff
@@ -0,0 +1,68 @@
+  end
+
+  it "can fetch service with distribution channel" do
+    service = Nurego::Service.retrieve({ id:'197f9207-7325-4d75-983a-116b43fb9cc8', distribution_channel:'website' })
```
and if I need to change the service id - i need to do it in multiple places...

bar-jakubovitz replied:
better to have it as a constant in the spec helper ?
